### DISCUSSION TITLE
Is there a way to skip packages during update the sig files of which are missing and that would prevent the update of all other packages?

### DISCUSSION THREAD

**Discussion by gc-user** [void-linux/void-packages#51292] [Q&A]
I often run into the problem, that because of some missing sig file for a package the update of all other packages fails, too. This would be not a big problem, if it only lasted a few minutes, as the the sig files are like bytes large and should synchronize in a very short time. But the same missing sig file is blocking the whole update process for hours at times.

I don't want to have to put the package on hold just to be able to update the rest of the packages that have an update.

I already checked
`xbps-install -h`

to see if maybe there's an option to skip the affected package, but I didn't see one.

So my question would be:

Is there an option that I missed or could an option be added that allows for skipping any package the sig file of which is still missing and that would hold up the update process?

**Comment by classabbyamp** [comment]
```
xbps-pkgdb -m hold somepkg
xbps-install -Su
xbps-pkgdb -m unhold somepkg
```
or just wait a bit

**Comment by gc-user** [comment]
Yeah, no, as I wrote: "I don't want to have to put the package on hold just to be able to update the rest of the packages that have an update."

For now, I resorted to switch repo mirrors (using your handy xmirror tool :-)) as they don't seem to be all at the same state, and thus either don't yet have some new packages or have the packages already synchronized with all sig files.

But an option for xbps-install would be much more elegant. Or maybe modify the sync tool of the repo servers to only put updated packages in the index if the sig file is also updated / put into the index.

**Comment by Chocimier** [comment]
> skip packages during update the sig files of which are missing

> don't want to have to put the package on hold

What would you want to do with them: 
- install without verifying signature?
- keep in old version, i.e. same as hold, but without explicit hold?
- other?

With what mirror this problem appear?

**Comment by gc-user** [comment]
Not sure if the second option is what I would "coin" "skip / ignore updates for packages for which the sig file is missing".

I use repo-de.voidlinux.org. So I can't tell if others also have that problem. If the issue occurs for that mirror I switch temporarily to repo-fi.voidlinux.org. So far, I haven't encountered that issue on the alternative repo, but I only used that one for the update at hand and switched back once the update was done, i.e. not long enough to really be able to tell either way. I only took notice once that on the fi repo the package in question wasn't updated yet, so there was chance to have a missing sig file for that particular package. But as that issue isn't package specific, that doesn't mean anything.
